src/js/hooks/cycleCount/useCountStepTable.jsx

```diff
@@ -40,6 +42,33 @@ const useCountStepTable = ({
     })();
   }, [currentLocation?.id]);
 
+  const groupBinLocationsByZone = () => {
+    const groupedByZone = binLocations.reduce((acc, bin) => {
+      const { zoneId, zoneName } = bin;
+      return {
+        ...acc,
+        [zoneId]: {
+          zoneId,
+          zoneName,
+          bins: [...(acc[zoneId]?.bins || []), {
+            id: bin.id,
+            name: bin.name,
+            label: bin.name,
+            value: bin.id,
+          }],
+        },
+      };
+    }, {});
+
+    return Object.values(groupedByZone).map((group) => ({
+      id: `zone-${group.zoneId}`,
+      name: group.zoneName,
+      label: <span className="zone-label">{group.zoneName}</span>,
+      isDisabled: true,
+      options: group.bins,
+    }));
+  };
```

Comment:
How is it done on the receiving page? Maybe you can reuse the code that was created before? If not, then please move this code to util files.